invenio_app_ils/ui/src/invenio_app_ils/common/api/loans/loan.js

```diff
@@ -0,0 +1,116 @@
+import { http } from '../base';
+import { toISO } from '../date';
+import { DateTime } from 'luxon';
+
+const loanURL = '/circulation/loans/';
+const get = loanPid => http.get(`${loanURL}${loanPid}`);
+const getLoanReplaceItemUrl = loanPid => `${loanURL}/${loanPid}/replace-item`;
+
+const postAction = (
+  url,
+  pid,
+  loan,
+  transactionUserPid,
+  transactionLocationPid
+) => {
+  const now = DateTime.local();
+  const payload = {
+    transaction_user_pid: transactionUserPid,
+    patron_pid: loan.patron_pid,
+    transaction_location_pid: transactionLocationPid,
+    transaction_date: toISO(now),
+  };
+
+  if ('item_pid' in loan) {
+    payload['item_pid'] = loan.item_pid;
+  } else if ('document_pid' in loan) {
+    payload['document_pid'] = loan.document_pid;
+  } else {
+    throw new Error(
+      `No 'item_pid' or 'document_pid' attached to loan '${pid}'`
+    );
+  }
+  return http.post(url, payload);
+};
+
+const assignItemToLoan = (itemPid, loanPid) => {
+  const url = getLoanReplaceItemUrl(loanPid);
+  const payload = { item_pid: itemPid };
+  return http.post(url, payload);
+};
+
+class Loans {
+  constructor(build) {
+    this.documentQuery = build.documentQuery ? build.documentQuery : '';
+    this.itemQuery = build.itemQuery ? build.itemQuery : '';
+    this.patronQuery = build.patronQuery ? build.patronQuery : '';
+    this.stateQuery = build.stateQuery ? build.stateQuery : '';
+  }
+
+  qs() {
+    return `${this.documentQuery}${this.itemQuery}${this.patronQuery}${
+      this.stateQuery
+    }`.replace(' AND ', '');
```

Comment:
this is only for the first `AND`. I chose to do it this way, because if I would use join I would have to check every query if it is not empty string and it produces more ifs

Reply:
What do you mean exactly?

```
var a = [];
var b = ["ciao"];
var c = ["mamma", "bla"];

a.join(' OR ')
""

b.join(' OR ')
"ciao"

c.join(' OR ')
"mamma OR bla"
``